### PR TITLE
Add MSRAPrelu initialization

### PR DESCRIPTION
**Pull request by danielgordon10**
Adds the initialization from "Delving Deep into Rectifiers: Surpassing Human-Level Performance on ImageNet Classification."

### CODE REVIEW

**Comment by antinucleon** [comment]
You can use ```Xavier(rnd_type="gaussain", factor_type="in", magnitude=2.)``` do exact same thing.

**Comment by danielgordon10** [comment]
That doesn't account for the slope. The paper recommends a starting slope of .25 for PReLUs, so it would be incorrect with Xavier.

**Comment by piiswrong** [comment]
Still I think you can use use `Xavier(rnd_type="gaussain", factor_type="in", magnitude=2./(1-slope*slope))

**Comment by danielgordon10** [comment]
That requires remembering a formula and introduces the possibility of errors.
It would be one thing if using Xavier made sense for any reason, but it seems to me like it was proved to be exponentially worse than this version. So it's not as if people should be using Xavier in the first place (but I know they still do).

**Comment by piiswrong** [comment]
In that case you can inherit Xavier. Less code to maintain

**Comment by antinucleon** [comment]
Also I don't think there is any use case for PReLU now.

**Comment by danielgordon10** [comment]
Fair point. Changed.
Why is there no use case for PReLU? I've had good success with it.

**Comment by antinucleon** [comment]
@danielgordon10 With BatchNorm, on ResNet or Inception, PReLU doesn't show any contribution in performace.

**Comment by antinucleon** [comment]
I disagree with that. Once the network starts converge, the difference of
final result will not be huge different. The key of Xavier family is keep
variance, so I can't see major difference.
On Mon, Jul 11, 2016 at 10:31 danielgordon10 <notifications@github.com>
wrote:

> @antinucleon <https://github.com/antinucleon> Even if that's the case, I
> still think this change is good because the initialization is more correct
> than Xavier even when using ReLU.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/dmlc/mxnet/pull/2647#issuecomment-231805497>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/ABM13qlGGp_O_ZIs2km5zuEQZbJxSRq-ks5qUn3rgaJpZM4JHgHe>
> .
>
-- 
Sent from mobile phone

**Comment by danielgordon10** [comment]
@antinucleon 
screenshot from 2016-07-11 13 48 01

**Comment by antinucleon** [comment]
In practice, I can't see any difference. The parameter ```magnitude``` is needed to be tuned at training. Both Xavier and Kaiming don't grantee the network will converge.

File: python/mxnet/initializer.py [hunk 1]
```diff
@@ -265,3 +268,19 @@ def _init_weight(self, _, arr):
...
+    ----------
+    factor_type: str, optional
+        Use ```avg```, ```in```, or ```out``` to init
+
+    slope: float, optional
+        initial slope of any PReLU (or similar) nonlinearities.
+    """
+    def __init__(self, factor_type="avg", slope=0.25):
+        magnitude = 2. / (1 + slope ** 2)
+        super(MSRAPrelu, self)__.init__("gaussian", factor_type, magnitude)
```
Comment (piiswrong) [inline python/mxnet/initializer.py]: Please fix gramma error and rebase. We can merge this in as long as it doesn't break existing code

Comment (danielgordon10) [inline python/mxnet/initializer.py]: What grammar error?

Comment (piiswrong) [inline python/mxnet/initializer.py]: __.init__ -> .__init__
This shouldn't be running. Have you tested it?

Comment (danielgordon10) [inline python/mxnet/initializer.py]: I tested it before I rewrote it... I guess that doesn't count.

**Comment by danielgordon10** [comment]
@antinucleon The fact is this is a three line addition of a convenience method that is more (theoretically) correct than the standard. I don't feel like I should have to fight this hard for it.

**Comment by danielgordon10** [comment]
Ping.

**Comment by piiswrong** [comment]
update branch and we can merge it

**Comment by danielgordon10** [comment]
Done.
